types/argparse/index.d.ts

```diff
@@ -79,13 +80,24 @@ export interface ArgumentGroupOptions {
     description?: string;
 }
 
+export abstract class Action {
+    protected dest: string;
+    constructor(options: ActionConstructorOptions);
+    abstract call(parser: ArgumentParser, namespace: Namespace, values: string[], optionString: string | null): void;
```

Comment:
Source code says `optionString` is an array: https://github.com/nodeca/argparse/blob/66012a0aadaa02572513c7b95b43d63f9753f678/lib/action.js#L133

Could you find source code that uses this param to verify?

Reply:
I believe the documentation in the source code is a copy/paste mistake.

The `ArgumentOptions` type has a `optionStrings` array, which includes the different ways to specify the argument `['-i', '--include']`.  But when `Action.call` is called, it is only passed the option that was actually used, and so it's a plain `string`.

Here's a short program that demonstrates it:

```
const argparse = require('argparse');

class MyAction extends argparse.Action {
    call(parser, namespace, values, optionString) {
        console.log(JSON.stringify({optionString, values}));
    }
}

const parser = argparse.ArgumentParser();
parser.addArgument(['-a'], {action: MyAction});
parser.addArgument(['-b', '-c'], {action: MyAction, nargs: '*'});
parser.parseArgs(['-a', 'x', '-b', 'y', '-c', 'z']);
```

Output:
```
{"optionString":"-a","values":"x"}
{"optionString":"-b","values":["y"]}
{"optionString":"-c","values":["z"]}
```

NOTE: When testing this, I noticed another issue.  My PR has the `values` parameter as an array, but it looks like it can sometimes be a string.  I've filed an issue (nodeca/argparse#135), let's wait and see what they say...
```